cmd/gitops/beta/run/cmd.go

```diff
@@ -50,6 +59,56 @@ func betaRunCommandPreRunE(endpoint *string) func(*cobra.Command, []string) erro
 
 func betaRunCommandRunE(opts *config.Options, client *resty.Client) func(*cobra.Command, []string) error {
 	return func(cmd *cobra.Command, args []string) error {
+		// If there is no cluster in the kube config, return an error.
+		cfg, clusterName, err := kube.RestConfig()
+		if err != nil {
+			return cmderrors.ErrNoCluster
+		}
+
+		// If there is a valid connection to a cluster when the command is run, connect to the currently selected cluster.
+		kubeClient, err := kube.NewKubeHTTPClientWithConfig(cfg, clusterName)
+		if err != nil {
+			return cmderrors.ErrGetKubeClient
+		}
+
+		log := internal.NewCLILogger(os.Stdout)
+
+		// Check if Flux is installed on the cluster.
+
+		ctx2 := context.Background()
+
+		fluxVersion, err := run.GetFluxVersion(log, ctx2, kubeClient)
+		if err != nil {
+			fmt.Println("error getting flux version", err)
+
+			fluxVersion = ""
+		}
+
+		fmt.Println("fluxVersion:", fluxVersion)
+
+		// If Flux is not installed on the cluster then the prerequisites will be installed to initiate the reconciliation process.
+		// This includes all default controllers to set up a reconciliation loop such as the notification-controller, helm-controller, kustomization-controller, and source-controller.
+		// This will also add all relevant CRDs from the controllers above such as Kustomizations, Helm Releases, Git Repository, Helm Repository, Bucket, Alerts, Providers, and Receivers.
+
+		if fluxVersion == "" {
+			filePath := args[0]
+
+			err = run.InstallFlux(log, ctx2, kubeClient, filepath.Join(filePath, run.FluxDirectory), run.ShortManifestFilename)
```

Comment:
I think we have to discuss the semantics of this flux location.

For example,
When running `gitops beta run ./deploy/overlay/dev`, the `filePath` variable would be `./deploy/overlay/dev`.

Then the `InstallFlux` function would install Flux manifests to `./deploy/overlay/dev/flux-system`, which would not be the correct location.

Per Flux convention, an on-disk location of Flux installation would be something like 
`./cluster/dev/flux-system` - which is normally passed to the `flux bootstrap` command.

So I think we need to refine this behavior.